src/test/java/com/exasol/bucketfs/SyncAwareBucketIT.java

```diff
@@ -219,19 +227,37 @@ void testInterruptedDuringUploadFile() throws BucketAccessException, Interrupted
     }
 
     @Test
-    void testIoExceptionDuringUploadFile() throws BucketAccessException, InterruptedException, TimeoutException, IOException {
+    void testIoExceptionDuringUploadFile()
+            throws BucketAccessException, InterruptedException, TimeoutException, IOException {
         final BucketAccessException exception = assertUploadThrowsExceptionWhenClientThrows(new IOException());
         assertThat(exception.getMessage(), Matchers.startsWith("E-BFSJ-7: I/O error trying to upload"));
     }
 
-    private BucketAccessException assertUploadThrowsExceptionWhenClientThrows(Exception exceptionThrownByHttpClient) throws InterruptedException, BucketAccessException, TimeoutException, IOException {
-        final var bucket = getDefaultBucket();
+    private BucketAccessException assertUploadThrowsExceptionWhenClientThrows(
+            final Exception exceptionThrownByHttpClient)
+            throws InterruptedException, BucketAccessException, TimeoutException, IOException {
+        final var bucket = createDefaultBucket();
         final String testFile = getUniqueFileName();
         bucket.uploadStringContent("test", testFile);
         final SyncAwareBucket bucketSpy = getBucketWithExceptionThrowingHttpClient(exceptionThrownByHttpClient, bucket);
         return assertThrows(BucketAccessException.class, () -> bucketSpy.uploadStringContent("test", testFile));
     }
 
+    @Test
+    void testUploadNecessityCheckStrategy(@TempDir final Path tempDir)
+            throws IOException, BucketAccessException, TimeoutException {
+        final UploadNecessityCheckStrategy uploadNecessityCheckStrategy = mock(UploadNecessityCheckStrategy.class);
```

Comment:
Extract mock creation and configuration to a method called `createUploadSwitchNecessityStrategy` parameterize with boolean and check both cases.